platform/switch/os_switch.cpp

```diff
@@ -379,14 +378,83 @@ void OS_Switch::key(uint32_t p_key, bool p_pressed) {
 	input->parse_input_event(ev);
 };
 
+//when only both joycon are use as a single controller (general case)
+std::vector< std::array<uint,2> > switch_joy_dual_button_map = {
+	{HidNpadButton_A, JOY_DS_A},
+	{HidNpadButton_B, JOY_DS_B},
+    {HidNpadButton_X, JOY_DS_X},
+    {HidNpadButton_Y, JOY_DS_Y},
+    {HidNpadButton_StickL, JOY_L3},
+    {HidNpadButton_StickR, JOY_R3},
+    {HidNpadButton_L, JOY_L},
+    {HidNpadButton_R, JOY_R},
+    {HidNpadButton_ZL, JOY_L2},
+    {HidNpadButton_ZR, JOY_R2},
+    {HidNpadButton_Plus, JOY_WII_PLUS},
+    {HidNpadButton_Minus, JOY_WII_MINUS},
+    {HidNpadButton_Left, JOY_DPAD_LEFT},
+    {HidNpadButton_Up, JOY_DPAD_UP},
+    {HidNpadButton_Right, JOY_DPAD_RIGHT},
+    {HidNpadButton_Down, JOY_DPAD_DOWN},
+};
+
+//when only right joycon is use as a controller horizontally
+std::vector< std::array<uint,2> > switch_joy_right_button_map = {
+	{HidNpadButton_A, JOY_DS_B},
+	{HidNpadButton_B, JOY_DS_Y},
+    {HidNpadButton_X, JOY_DS_A},
+    {HidNpadButton_Y, JOY_DS_X},
+    {HidNpadButton_ZR, JOY_R2},
+    {HidNpadButton_Plus, JOY_START},
+    {HidNpadButton_StickRLeft, JOY_DPAD_UP},
+    {HidNpadButton_StickRUp, JOY_DPAD_RIGHT},
+    {HidNpadButton_StickRRight, JOY_DPAD_DOWN},
+    {HidNpadButton_StickRDown, JOY_DPAD_LEFT},
+    {HidNpadButton_RightSL, JOY_L},
+    {HidNpadButton_RightSR, JOY_R}
+};
+
+//when only left joycon is use as a controller horizontally
+std::vector< std::array<uint,2> > switch_joy_left_button_map = {
+    {HidNpadButton_ZL, JOY_L2},
+    {HidNpadButton_Minus, JOY_START},
+    {HidNpadButton_Left, JOY_DS_B},
+    {HidNpadButton_Up, JOY_DS_Y},
+    {HidNpadButton_Right, JOY_DS_X},
+    {HidNpadButton_Down, JOY_DS_A},
+    {HidNpadButton_StickLLeft, JOY_DPAD_DOWN},
+    {HidNpadButton_StickLUp, JOY_DPAD_LEFT},
+    {HidNpadButton_StickLRight, JOY_DPAD_UP},
+    {HidNpadButton_StickLDown, JOY_DPAD_RIGHT},
+    {HidNpadButton_LeftSL, JOY_L},
+    {HidNpadButton_LeftSR, JOY_R}
+};
+
+struct PadConfiguration {
+	bool initialized = false;
+	std::vector<std::array<uint,2>> mapping = {};
+};
+
 void OS_Switch::run() {
+
 	if (!main_loop) {
-		TRACE("no main loop???\n");
+		ERR_FAIL_MSG("no main loop");
 		return;
 	}
 
 	main_loop->init();
 
+    padConfigureInput(4, HidNpadStyleSet_NpadStandard);
```

Comment:
`4` here and in other places should ideally be changed to a named constant, like `#define MAX_JOYPADS 4`

Reply:
Done ;)